chatgpt_interface/chat_gpt_client.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+from openai_api import OpenAIChatAPI
+
+
+async def is_answer_relevant(chat: OpenAIChatAPI, question: str, answer: str) -> bool:
+    """
+    Determines whether an answer to a given question is relevant using the OpenAIChatAPI.
+
+    Args:
+        chat (OpenAIChatAPI): An instance of the OpenAIChatAPI class for communication with OpenAI.
+        question (str): The question for which the answer's relevance needs to be assessed.
+        answer (str): The answer to be evaluated for relevance.
+
+    Returns:
+        bool: True if the answer is relevant, False otherwise.
+    """
+    prompt = f"Please provide a concise response: Is the answer relevant? (Yes/No)\n\nQuestion: {question}\nAnswer: {answer}\n"
+    response = await chat.generate_response(prompt)
+    return response.lower() == 'yes'
+
+
+async def calculate_scores(chat: OpenAIChatAPI, questions_and_answers):
+    """
+    Calculates scores for doctor's behavior and professionalism based on patient feedback.
+
+    Args:
+        chat (OpenAIChatAPI): An instance of the OpenAIChatAPI class for communication with OpenAI.
+        questions_and_answers (list): A list of tuples, each containing a question and its corresponding answer.
+
+    Returns:
+        str: A double representing the calculated scores.
+    """
+    score = 0
+    for question, answer in questions_and_answers:
```

Comment:
It is not possible to send several requests together in the free version of api key